SparkConnector/sparkconnector/configuration.py

```diff
@@ -132,11 +132,27 @@ def configure(self, opts, ports):
 
         # Extend conf adding logging of log4j to java options
         base_extra_java_options = "-Dlog4j.configuration=file:%s" % self.connector.log4j_file
-        extra_java_options = conf.get("spark.driver.extraJavaOptions")
-        if extra_java_options:
-            extra_java_options = base_extra_java_options + " " + extra_java_options
+        extra_java_options = None
+
+        # For customenv sessions, do not allow extraJavaOptions to be overwritten by the extra_java_options, 
+        # since the NXCals instances, coming as a default value by the SparkContext when extraJavaOptions is
+        # not specified, need to be preserved in the driver configuration.
+        if os.environ.get("SOFTWARE_SOURCE") == "customenv":
+            SPARK_DEFAULTS = os.environ.get(
+                "SPARK_DEFAULTS", 
+                os.path.join(os.environ.get("SPARK_HOME"), "conf", "spark-defaults.conf")
+            )
+            if os.path.exists(SPARK_DEFAULTS):
```

Comment:
We need an else here: if the file does not exist, we just set `extra_java_options = base_extra_java_options`, as it is done in the case of the LCG release below.